internal/pkg/name-conversions/name-conversions.go

```diff
@@ -0,0 +1,97 @@
+	if err != nil {
+		return input, err
+	}
+	clusterPtrs := ConvertToPtrSlice(clusters)
+	specPtrs := make([]*cmkv2.CmkV2ClusterSpec, len(clusters))
+	for i := range clusters {
+		specPtrs[i] = clusterPtrs[i].Spec
+	}
+	return ConvertSpecNameToId(input, clusterPtrs, specPtrs)
+}
+
+// EnvironmentNameToId attempts to convert from a valid environment name to its ID, if it fails it returns an error
+func EnvironmentNameToId(input string, v2Client *ccloudv2.Client, tryGetFirst bool) (string, error) {
+	if tryGetFirst {
+		if env, err := v2Client.GetOrgEnvironment(input); err == nil {
+			return env.GetId(), err
+		}
+	}
+	envs, err := v2Client.ListOrgEnvironments()
+	if err != nil {
+		return input, err
+	}
+	return ConvertV2NameToId(input, ConvertToPtrSlice(envs))
+}
+
+// IamPoolNameToId attempts to convert from a valid iam pool name to its ID, if it fails it returns an error
+func IamPoolNameToId(input string, providerId string, v2client *ccloudv2.Client, tryGetFirst bool) (string, error) {
+	if tryGetFirst {
+		if pool, err := v2client.GetIdentityPool(input, providerId); err == nil {
+			return pool.GetId(), err
+		}
+	}
+	pools, err := v2client.ListIdentityPools(providerId)
+	if err != nil {
+		return input, err
+	}
+	return ConvertV2NameToId(input, ConvertToPtrSlice(pools))
+}
+
+// IamProviderNameToId attempts to convert from a valid iam provider name to its ID, if it fails it returns an error
+func IamProviderNameToId(input string, v2client *ccloudv2.Client, tryGetFirst bool) (string, error) {
+	if tryGetFirst {
+		if provider, err := v2client.GetIdentityProvider(input); err == nil {
+			return provider.GetId(), err
+		}
+	}
+	providers, err := v2client.ListIdentityProviders()
+	if err != nil {
+		return input, err
+	}
+	return ConvertV2NameToId(input, ConvertToPtrSlice(providers))
+}
+
+// IamServiceAccountNameToId attempts to convert from a valid iam service account name to its ID, if it fails it returns an error
+func IamServiceAccountNameToId(input string, v2client *ccloudv2.Client, tryGetFirst bool) (string, error) {
+	if tryGetFirst {
+		if serviceAccount, _, err := v2client.GetIamServiceAccount(input); err == nil {
+			return serviceAccount.GetId(), err
+		}
+	}
+	serviceAccounts, err := v2client.ListIamServiceAccounts()
+	if err != nil {
+		return input, err
+	}
+	return ConvertV2NameToId(input, ConvertToPtrSlice(serviceAccounts))
+}
+
+// QuotaNameToId attempts to convert from a valid quota name to its ID, if it fails it returns an error
+func QuotaNameToId(input string, clusterId string, environmentId string, v2Client *ccloudv2.Client) (string, error) {
+	quotas, err := v2Client.ListKafkaQuotas(clusterId, environmentId)
+	if err != nil {
+		return input, err
+	}
+	quotaPtrs := ConvertToPtrSlice(quotas)
+	specPtrs := make([]*v1.KafkaQuotasV1ClientQuotaSpec, len(quotas))
+	for i := range quotas {
+		specPtrs[i] = quotaPtrs[i].Spec
+	}
+	return ConvertSpecNameToId(input, quotaPtrs, specPtrs)
```

Comment:
What's going on here? Are any of these conversions unecessary?

Reply:
I have to take the pointers of the resource `specs` because their `GetId()` function takes a pointer receiver.